commons/src/y-doc-sync/realtime-doc.ts

```diff
@@ -3,27 +3,53 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
-import { Doc } from 'yjs'
-import { Text as YText } from 'yjs'
+import { EventEmitter2 } from 'eventemitter2'
+import type { EventMap } from 'eventemitter2'
+import {
+  applyUpdate,
+  Doc,
+  encodeStateAsUpdate,
+  encodeStateVector,
+  Text as YText
+} from 'yjs'
 
 const MARKDOWN_CONTENT_CHANNEL_NAME = 'markdownContent'
 
+export interface RealtimeDocEvents extends EventMap {
+  update: (update: number[], origin: unknown) => void
+}
+
 /**
  * This is the implementation of {@link Doc YDoc} which includes additional handlers for message sending and receiving.
  */
-export class RealtimeDoc extends Doc {
+export class RealtimeDoc extends EventEmitter2<RealtimeDocEvents> {
+  private doc: Doc = new Doc()
+  private readonly docUpdateListener: (
+    update: Uint8Array,
+    origin: unknown
+  ) => void
+
   /**
    * Creates a new instance.
    *
    * The new instance is filled with the given initial content.
    *
    * @param initialContent - the initial content of the {@link Doc YDoc}
    */
-  constructor(initialContent?: string) {
+  constructor(initialContent?: string | number[]) {
     super()
     if (initialContent) {
-      this.getMarkdownContentChannel().insert(0, initialContent)
+      if (typeof initialContent === 'string') {
```

Comment:
I'm a bit unhappy with this union type and the following `typeof`.

Actually, with the whole concept of mixing "a string of initial content" with "an arbitrary binary blob of yjs state" that gets introduced with this PR. I think these two types of data should be more cleanly separated.